WalletWasabi.Fluent/ViewModels/Wallets/Send/ChangeAvoidanceSuggestionViewModel.cs

```diff
@@ -0,0 +1,155 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.Blockchain.TransactionBuilding;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Send;
+
+public partial class ChangeAvoidanceSuggestionViewModel : SuggestionViewModel
+{
+	[AutoNotify] private string _amount;
+	[AutoNotify] private string _amountFiat;
+	[AutoNotify] private List<PrivacySuggestionBenefit> _benefits;
+	[AutoNotify] private PrivacyOptimisationLevel _optimisationLevel;
+	[AutoNotify] private bool _optimisationLevelGood;
+
+	public ChangeAvoidanceSuggestionViewModel(decimal originalAmount,
+		BuildTransactionResult transactionResult,
+		PrivacyOptimisationLevel optimisationLevel,
+		decimal fiatExchangeRate,
+		bool isOriginal,
+		params PrivacySuggestionBenefit[] benefits) : base(transactionResult, isOriginal)
+	{
+		_optimisationLevel = optimisationLevel;
+		_benefits = benefits.ToList();
+
+		decimal total = transactionResult.CalculateDestinationAmount().ToDecimal(MoneyUnit.BTC);
+
+		var fiatTotal = total * fiatExchangeRate;
+
+		_amountFiat = total.GenerateFiatText(fiatExchangeRate, "USD");
+		_optimisationLevelGood = optimisationLevel == PrivacyOptimisationLevel.Better;
+
+		if (_optimisationLevelGood)
+		{
+			var fiatOriginal = originalAmount * fiatExchangeRate;
+			var fiatDifference = fiatTotal - fiatOriginal;
+
+			var difference = (fiatDifference > 0
+					? $"{fiatDifference.GenerateFiatText("USD")} More"
+					: $"{Math.Abs(fiatDifference).GenerateFiatText("USD")} Less")
+				.Replace("(", "").Replace(")", "");
+
+			_benefits.Add(new(false, difference));
+		}
+		else
+		{
+			// This is just to pad the control.
+			_benefits.Add(new(false, " "));
+		}
+
+		_amount = $"{total}";
+	}
+
+	private static IEnumerable<ChangeAvoidanceSuggestionViewModel> NormalizeSuggestions(
+		IEnumerable<ChangeAvoidanceSuggestionViewModel> suggestions,
+		ChangeAvoidanceSuggestionViewModel defaultSuggestion)
+	{
+		var normalized = suggestions
+			.OrderBy(x => x.TransactionResult.CalculateDestinationAmount())
+			.ToList();
+
+		if (normalized.Count == 3)
+		{
+			var index = normalized.IndexOf(defaultSuggestion);
+
+			switch (index)
+			{
+				case 1:
```

Comment:
Why not keep the correct order of the cases to be easy to read the code?

Reply:
This code removes the duplications. If two "pay less" or "pay more" was generated. One is fairly enough.

Reply:
not part of this PR.. just moved.